src/awx_plugins/credentials/github_app.py

```diff
@@ -0,0 +1,190 @@
+)
+from awx_plugins.interfaces._temporary_private_django_api import (  # noqa: WPS436
+    gettext_noop as _,
+)
+
+from github import Auth, Github
+
+from .plugin import CredentialPlugin
+
+
+class FieldDict(TypedDict):
+    id: str
+    label: str
+    type: str
+    format: NotRequired[str]
+    secret: NotRequired[bool]
+    multiline: NotRequired[bool]
+    help_text: str
+    default: str
+
+
+class MetadataDict(TypedDict):
+    id: str
+    label: NotRequired[str]
+    type: str
+    help_text: str
+
+
+class GitHubAppInputs(TypedDict):
+    fields: list[FieldDict]
+    metadata: list[MetadataDict]
+    required: list[str]
+
+
+github_app_inputs: GitHubAppInputs = {
+    'fields': [
+        {
+            'id': 'github_url',
+            'label': _('GitHub URL'),
+            'type': 'string',
+            'help_text': _(
+                'Specify the GitHub URL here in the case of an Enterprise Github. '
+                'Example: https://gh.your.org/api/v3',
+            ),
+            'default': 'https://api.github.com',
+        },
+        {
+            'id': 'app_id',
+            'label': _('GitHub App id created by the GitHub Admin'),
+            'type': 'string',
+            'help_text': _(
+                'The GitHub App ID created by the GitHub Admin. '
+                'Example App ID: 1121547 '
+                'from https://github.com/settings/apps/arestlel-github-app-ansible '
+                'required for creating a JWT token for authentication',
+            ),
+        },
+        {
+            'id': 'install_id',
+            'label': _('Installation ID'),
+            'type': 'string',
+            'help_text': _(
+                'The Installation ID from the GitHub App installation generated by the GitHub Admin. '
+                'Example: 59980338 '
+                'extracted from the installation link: https://github.com/settings/installations/59980338 '
+                'required for creating a limited GitHub app token',
+            ),
+        },
+        {
+            'id': 'private_rsa_key',
+            'label': _('RSA Private Key'),
+            'type': 'string',
+            'format': 'ssh_private_key',  # this validator checks for formats close to PKCS#1 PEMs
+            'secret': True,
+            'multiline': True,
+            'help_text': _(
+                'Paste the contents of the PEM file that the GitHub Admin '
+                'provided to you with the app and installation IDs.',
+            ),
+        },
+    ],
+    'metadata': [
+        {
+            'id': 'description',
+            'label': _('Description (Optional)'),
+            'type': 'string',
+            'help_text': _('To be removed after UI is updated'),
+        },
+    ],
+    'required': ['app_id', 'install_id', 'private_rsa_key'],
+}
+
+class GitHubAppBackendArgs(TypedDict, total=False):
+    github_url: str
+    app_id: str | int
+    install_id: str | int
+    private_rsa_key: str
+    jwt_expiry: str | int
```

Comment:
Looking into https://github.com/PyGithub/PyGithub/blob/d90323f/github/Auth.py#L197C9-L197C53 and taking into account that JWT expiry isn't configurable through `github_app_inputs`, I realized that it's not actually set anywhere. And I don't think it needs to be, as the default is good.
So let's just drop it to avoid maintaining yet another thing.